tests/run/i505.scala

```diff
@@ -0,0 +1,15 @@
+object Test {
+  def main(args: Array[String]): Unit = {
+    val a: Int = synchronized(1)
```

Comment:
Would be nice to have a bytecode test that verifies we never emit a call to a boxing or unboxing method.